docs/howto/send-notifications.rst

```diff
@@ -0,0 +1,189 @@
+Send Notifications
+==================
+
+Taskcluster can `send notifications`_ when a task transitions states.
+
+Notification Types
+------------------
+
+There are four supported notification types:
+
+1. Email - An email can be sent to any address.
+2. Pulse - A `pulse event`_ can be triggered.
+3. Matrix - A Matrix room can be pinged. Requires the Taskcluster deployment to
+   be configured with Matrix credentials and be invited to the room.
+4. Slack - A slack channel can be pinged. Requires the Taskcluster deployment to
+   be configured with Slack credentials and be invited to the room.
+
+.. _status-types:
+
+Status Types
+------------
+
+A notification can be sent on the following task status changes:
+
+- ``on-defined`` - When the task is first submitted to Taskcluster.
```

Comment:
Good call on not supporting the deprecated `on-any`[1]:

> **on-any:** (DEPRECATED) This does what it sounds like and will notify you of task-completion no matter how it ends. Unless the task was deliberately canceled.

https://docs.taskcluster.net/docs/reference/core/notify/usage#filters